tfx/examples/imdb/imdb_utils_native_keras.py

```diff
@@ -0,0 +1,215 @@
+  """
+  transformed_feature_spec = (
+      tf_transform_output.transformed_feature_spec().copy())
+
+  dataset = tf.data.experimental.make_batched_features_dataset(
+      file_pattern=file_pattern,
+      batch_size=batch_size,
+      features=transformed_feature_spec,
+      reader=_gzip_reader_fn,
+      label_key=_transformed_name(_LABEL_KEY))
+
+  return dataset
+
+def _build_keras_model() -> keras.Model:
+  """Creates a LSTM Keras model for classifying imdb data.
+  Reference: https://www.tensorflow.org/tutorials/text/text_classification_rnn
+
+  Returns:
+    A Keras Model.
+  """
+  # The model below is built with Sequential API, please refer to
+  # https://www.tensorflow.org/guide/keras/sequential_model
+  model = keras.Sequential([
+      keras.layers.Embedding(
+          _VOCAB_SIZE+1,
+          _EMBEDDING_UNITS,
+          name=_transformed_name(_FEATURE_KEY)
+          ),
+      keras.layers.Bidirectional(keras.layers.LSTM(
+          _LSTM_UNITS,
+          dropout=_DROPOUT_RATE)),
+      keras.layers.Dense(_HIDDEN_UNITS, activation='relu'),
+      keras.layers.Dense(1)
+  ])
+
+  model.compile(
+      loss=keras.losses.BinaryCrossentropy(from_logits=True),
+      optimizer=keras.optimizers.Adam(_LEARNING_RATE),
+      metrics=['accuracy'])
+
+  model.summary(print_fn=absl.logging.info)
+  return model
+
+def _get_serve_tf_examples_fn(model, tf_transform_output):
+  """Returns a function that parses a serialized tf.Example."""
+  model.tft_layer = tf_transform_output.transform_features_layer()
+
+  @tf.function
+  def serve_tf_examples_fn(serialized_tf_examples):
+    """Returns the output to be used in the serving signature."""
+    feature_spec = tf_transform_output.raw_feature_spec()
+    feature_spec.pop(_LABEL_KEY)
+    parsed_features = tf.io.parse_example(serialized_tf_examples, feature_spec)
+    transformed_features = model.tft_layer(parsed_features)
+    return model(transformed_features)
+
+  return serve_tf_examples_fn
+
+
+# TFX Trainer will call this function.
+def run_fn(fn_args: TrainerFnArgs):
+  """Train the model based on given args.
+
+  Args:
+    fn_args: Holds args used to train the model as name/value pairs.
+  """
+  tf_transform_output = tft.TFTransformOutput(fn_args.transform_output)
+
+  train_dataset = _input_fn(
+      fn_args.train_files,
+      tf_transform_output,
+      batch_size=_TRAIN_BATCH_SIZE)
+
+  eval_dataset = _input_fn(
+      fn_args.eval_files,
+      tf_transform_output,
+      batch_size=_EVAL_BATCH_SIZE)
+
+  mirrored_strategy = tf.distribute.MirroredStrategy()
+  with mirrored_strategy.scope():
+    model = _build_keras_model()
+
+  # In distributed training, it is common to use num_steps instead of num_epochs
+  # to control training.
+  # Reference: https://stackoverflow.com/questions/45989971/
+  # /distributed-training-with-tf-estimator-resulting-in-more-training-steps
+
+  model.fit(
+      train_dataset,
+      epochs=1,
```

Comment:
this is default if I remember correctly, don't need to set it

Reply:
Yes it is.
fixed